crates/pera-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -485,17 +490,56 @@ impl DWalletMPCManager {
         let handle = tokio::runtime::Handle::current();
         let session = session.clone();
         let finished_computation_sender = self.completed_computation_channel_sender.clone();
-        rayon::spawn_fifo(move || {
-            if let Err(err) = session.advance(&handle) {
-                error!("Failed to advance session with error: {:?}", err);
-            }
-            if let Err(err) = finished_computation_sender.send(()) {
-                error!(
-                    "Failed to send finished computation message with error: {:?}",
-                    err
-                );
-            }
-        });
+        if matches!(
+            session.session_info.mpc_round,
+            MPCInitProtocolInfo::Sign(..)
+        ) && session.pending_quorum_for_highest_round_number == LAST_SIGN_ROUND_INDEX
+        {
+            let sign_last_step_delay =
+                self.calculate_last_sign_step_validator_delay(&session.session_info)?;
+            let epoch_store = self.epoch_store()?;
+            tokio::spawn(async move {
+                tokio::time::sleep(tokio::time::Duration::from_secs(
+                    sign_last_step_delay as u64,
+                ))
+                .await;
+                let manager = epoch_store.get_dwallet_mpc_manager().await;
+                let Some(session) = manager.mpc_sessions.get(&session_id) else {
+                    error!("failed to get session when checking if sign last round should get executed");
+                    return;
+                };
+                if session.status == MPCSessionStatus::Active {
+                    info!(
+                        "running last sign cryptographic step for session_id: {:?}",
+                        session_id
+                    );
+                    let session = session.clone();
+                    rayon::spawn_fifo(move || {
+                        if let Err(err) = session.advance(&handle) {
+                            error!("Failed to advance session with error: {:?}", err);
+                        }
+                        if let Err(err) = finished_computation_sender.send(()) {
+                            error!(
+                                "Failed to send finished computation message with error: {:?}",
+                                err
+                            );
+                        }
+                    });
+                }
+            });
```

Comment:
move this to a func, it doesn't really look good when we have a specific case in the general code, hard to follow.